creds/cred.go

```diff
@@ -0,0 +1,76 @@
+package vaultlib
```

Comment:
make the name of package, file and directory the same. I suggest to name it "vault".

create a sub directory under vault. "vault/hashicorp/hashicorp.go" will create an object which satisfies vault interface.